test/files/neg/caseclass_private_constructor.scala

```diff
@@ -1,4 +1,4 @@
-// scalac: -Xsource:2.14
+// scalac: -Xsource:3
```

Comment:
Considering Scala 3.0 and 3.1 might deprecate different things, we might want to stick to two digit `-Xsource:3.0`?

Reply:
Also given that `-Xsource:2.14` is an `-X` option, doesn't the unwritten deprecation [policy](https://github.com/scala/scala/pull/7686#issuecomment-464753832) stipulate that we can't remove its support until Scala 3.1 comes out?

Reply:
I would consider that policy relevant to `-Xsource` as a whole, but I don't see a problem with `-Xsource:2.14` as a particular sub-option

Reply:
-Xsource:3 means 3.0.  Actually it means 3.0.0 (and there's suffix support too).  It's the lower bound.

Reply:
If someone compiled using `-Xsource:2.14` and got some sort of semantics (like no procedure syntax), shouldn't the semantics be preserved throughout the 2.13.x cycle?

Reply:
IMO no, because `-Xsource` is informational, not something you leave enabled, as per https://github.com/scala/bug/issues/11661

Reply:
That makes it sound like `-Xsource` should be `-Ysource`.

So we can do that (via a deprecation cycle; though Martin in lampepfl/dotty#8571 is thinking of using `-source`).

Or maybe (assuming it works) we can try to make `-Xsource:2.14` do a postSetHook to set itself to 3 instead, and ideally emit a deprecation warning too.

Or just ignore the policy.

Reply:
> we can try to make -Xsource:2.14 do a postSetHook to set itself to 3 instead, and ideally emit a deprecation warning too

that would be fine with me